src/keybindings.c

```diff
@@ -1122,6 +1129,18 @@ static gboolean check_fixed_kb(guint keyval, guint state)
 			return TRUE;
 		}
 	}
+	/* temporarily show the menubar again when triggering it while hidden */
+	if (state == 0 && keyval == GDK_KEY_F10 && ! ui_prefs.menubar_visible)
+	{
+		GtkWidget *const geany_menubar_box = ui_lookup_widget(main_widgets.window, "hbox_menubar");
+		GtkWidget *const geany_menubar = ui_lookup_widget(main_widgets.window, "menubar1");
+
+		gtk_widget_show(geany_menubar_box);
+		gtk_menu_shell_select_first(GTK_MENU_SHELL(geany_menubar), TRUE);
+
+		return TRUE;
+	}
+	/* FIXME: handle Alt to focus the menu bar? */
```

Comment:
Maybe we should remove this FIXME, or at least make it a TODO or something less critical?
```suggestion
```